### PR TITLE
Set Custom Resource Names for Azure-Arm builds

### PR DESCRIPTION
**Pull request by vijayrajah**
This PR enables the option to specify custom resource name prefix during the build.

This introduces one new config parameter -- custom_resource_build_prefix (max of 10 characters)

If this option is set, then the resource names would be `custom_resource_build_prefix` + resourceType + 5 random characets
EX:

if custom_resource_build_prefix = "centos7bld", then VM name becomes centos7bldvm+5 Random Alphanum chars.

### CODE REVIEW

File: builder/azure/arm/tempname.go [hunk 1]
```diff
@@ -23,21 +23,36 @@ type TempName struct {
...
+		tempName.KeyVaultName = fmt.Sprintf("%skv%s", p, suffix)
+		tempName.OSDiskName = fmt.Sprintf("%sos%s", p, suffix)
+		tempName.NicName = fmt.Sprintf("%sni%s", p, suffix)
+		tempName.PublicIPAddressName = fmt.Sprintf("%sip%s", p, suffix)
+		tempName.SubnetName = fmt.Sprintf("%ssn%s", p, suffix)
+		tempName.VirtualNetworkName = fmt.Sprintf("%svn%s", p, suffix)
+		tempName.NsgName = fmt.Sprintf("%ssg%s", p, suffix)
+		tempName.ResourceGroupName = fmt.Sprintf("%s-Resource-Group-%s", p, suffix)
+
+	}
```
Comment (sylviamoss) [inline builder/azure/arm/tempname.go:55]: I think you could just do it like:
```suggestion
	suffix := random.AlphaNumLower(5)
	if p == "" {
	    p = "pkr"
	    suffix = random.AlphaNumLower(10)
	} 
	tempName.ComputeName = fmt.Sprintf("%svm%s", p, suffix)
	tempName.DeploymentName = fmt.Sprintf("%sdp%s", p, suffix)
	tempName.KeyVaultName = fmt.Sprintf("%skv%s", p, suffix)
	tempName.OSDiskName = fmt.Sprintf("%sos%s", p, suffix)
	tempName.NicName = fmt.Sprintf("%sni%s", p, suffix)
	tempName.PublicIPAddressName = fmt.Sprintf("%sip%s", p, suffix)
	tempName.SubnetName = fmt.Sprintf("%ssn%s", p, suffix)
	tempName.VirtualNetworkName = fmt.Sprintf("%svn%s", p, suffix)
	tempName.NsgName = fmt.Sprintf("%ssg%s", p, suffix)
	tempName.ResourceGroupName = fmt.Sprintf("%s-Resource-Group-%s", p, suffix)
```

File: builder/azure/arm/tempname_test.go [hunk 2]
```diff
@@ -101,3 +101,43 @@ func TestTempNameShouldHaveSameSuffix(t *testing.T) {
...
+	}
+
+	if strings.Index(tempName.VirtualNetworkName, "CustPrefixvn") != 0 {
+		t.Errorf("Expected VirtualNetworkName to begin with 'CustPrefixvn', but got '%s'!", tempName.VirtualNetworkName)
+	}
+
+	if strings.Index(tempName.NsgName, "CustPrefixsg") != 0 {
+		t.Errorf("Expected NsgName to begin with 'CustPrefixsg', but got '%s'!", tempName.NsgName)
+	}
+}
```
Comment (sylviamoss) [inline builder/azure/arm/tempname_test.go:143]: nice!

**Comment by sylviamoss** [comment]
I tested it locally and it works 👍

File: builder/azure/arm/tempname_test.go (hunk 2 above)
Comment (vijayrajah) [reply builder/azure/arm/tempname_test.go:143]: Thanks..

**Comment by vijayrajah** [comment]
@sylviamoss .. Thanks for the feedback.. I have incorporated the changes..

**Pull request merged**